hs-ucan/library/Web/UCAN/Resolver/Class.hs

```diff
@@ -0,0 +1,11 @@
+module Web.UCAN.Resolver.Class (Resolver (..)) where
+
+import           Network.IPFS.CID.Types
+
+import           RIO
+
+import qualified Web.UCAN.RawContent     as UCAN
+import           Web.UCAN.Resolver.Error as Resolver
+
+class Monad m => Resolver m where
+  resolve :: CID -> m (Either Resolver.Error UCAN.RawContent)
```

Comment:
Here's what I've been talking about (above?): The `Resolver` class now returns
`Either Resolver.Error UCAN.RawContent` instead of
`Either Resolver.Error (UCAN.RawContent, UCAN.UCAN fct ptc rsc)`,
which just makes it a little simpler by getting rid of some dependencies on what `fct`, `ptc` and `rsc` are.